### DISCUSSION TITLE
Authentication / Access-control best practice?

### DISCUSSION THREAD

**Discussion by do-me** [qdrant/qdrant#1418] [General]
Are there any plans yet to implement authentication or role-based access control or similar? 
If not, are there any guidelines or a best practice example of how to possibly set it up with docker? 

I would like to differentiate between: 
- read-only users (customers or general public)
- users with write permissions (dev)
and if possible for each collection separately.

Row-level security would also be nice, but the basics above would also do for the beginning.

**Comment by sbusso** [comment]
Coming here as I was looking at the same question. Have you implemented any auth @do-me ?

**Reply by do-me** [reply]
No, I am not working on it. The discussions are mainly on discord if you would like to push the issue https://github.com/qdrant/qdrant/issues/1499.

**Comment by brandonbiggs** [comment]
Hi, I wanted to see if there's been any update? I'm also interested in collection specific permissions/RBAC.

**Comment by azayarni** [comment]
We are working on it as a part of the cloud/on-prem solution.
PS: We do answer all questions on Discord https://qdrant.to/discord

**Reply by brandonbiggs** [reply]
Thanks @azayarni ! Is the on-prem solution part of the enterprise version or the community version as well?

**Reply by azayarni** [reply]
Part of Cloud/Enterprise version.

**Comment by azayarni** [comment]
Update. RBAC will be available in the upcoming 1.9 version.

**Reply by brandonbiggs** [reply]
Still as part of the enterprise/cloud version I assume?

**Reply by generall** [reply]
no, it will be open-source as well

**Reply by brandonbiggs** [reply]
This is so exciting! Thank you and the qdrant team for all the hard work!

**Reply by Dojchinovska** [reply]
Hello, any updates regarding this? When is the next release scheduled for? Thank you

**Reply by timvisee** [reply]
Our current estimation is somewhere next week.

**Reply by azayarni** [reply]
The release is planned for next week.

**Reply by Dojchinovska** [reply]
Exciting! Thank you very much for your fast response, great job!
